test/extended/builds/volumes.go

```diff
@@ -0,0 +1,130 @@
+package builds
+
+import (
+	"path/filepath"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	deploymentutil "github.com/openshift/origin/test/extended/deployments"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-builds][Feature:Builds][volumes] build volumes", func() {
+	var (
+		oc                     = exutil.NewCLI("build-volumes")
+		baseDir                = exutil.FixturePath("testdata", "builds", "volumes")
+		secret                 = filepath.Join(baseDir, "secret.yaml")
+		configmap              = filepath.Join(baseDir, "configmap.yaml")
+		s2iImageStream         = filepath.Join(baseDir, "s2i-imagestream.yaml")
+		s2iBuildConfig         = filepath.Join(baseDir, "s2i-buildconfig.yaml")
+		s2iDeploymentConfig    = filepath.Join(baseDir, "s2i-deploymentconfig.yaml")
+		dockerImageStream      = filepath.Join(baseDir, "docker-imagestream.yaml")
+		dockerBuildConfig      = filepath.Join(baseDir, "docker-buildconfig.yaml")
+		dockerDeploymentConfig = filepath.Join(baseDir, "docker-deploymentconfig.yaml")
+	)
+
+	g.Context("", func() {
+
+		g.BeforeEach(func() {
+			exutil.PreTestDump()
+		})
+
+		g.JustBeforeEach(func() {
+			g.By("waiting for openshift namespace imagestreams")
+
+			g.By("creating a secret")
+			err := oc.Run("create").Args("-f", secret).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("creating a configmap")
+			err = oc.Run("create").Args("-f", configmap).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpConfigMapStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+			}
+		})
+
+		g.It("should mount given secrets and configmaps into the build pod for source strategy builds", func() {
```

Comment:
This test will not work for disconnected builds (e2e-metal-ipi-ipv6), so it must be added in the Disconnected list here: https://github.com/openshift/origin/blob/b418b4058c91c873b5a2cc180c8718686793c4db/test/extended/util/annotate/rules.go#L91

Reply:
added